app/src/main/java/com/termux/api/apis/BluetoothScanAPI.java

```diff
@@ -0,0 +1,122 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BluetoothScanAPI {
+
+    private static final ArrayList<BluetoothDevice> devices = new ArrayList<>();
+    private static final String LOG_TAG = "BluetoothScanAPI";
+    private static BluetoothAdapter adapter;
+    private static boolean scanning = false;
+
+    private static final BroadcastReceiver receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (Objects.equals(intent.getAction(), BluetoothDevice.ACTION_FOUND)) {
+                BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
+                if (device != null) {
+                    devices.add(device);
+                }
+            }
+        }
+    };
+
+    public static void onReceive(final Context context, final Intent intent) {
+        Logger.logDebug(LOG_TAG, "onReceive");
+
+        ResultReturner.returnData(context, intent, new ResultReturner.ResultJsonWriter() {
+            @Override
+            public void writeJson(JsonWriter out) throws Exception {
+                String mode = intent.getStringExtra("mode");
+                if (mode == null) return;
+                out.beginObject();
+                switch (mode.toLowerCase()) {
+                    case "start":
+                        if (start(context)) {
+                            out.name("error").value(false);
+                        } else {
+                            out.name("error").value(true);
+                            out.name("reason").value("Already running!");
+                        }
+                        break;
+                    case "stop":
+                        if (stop(context)) {
+                            out.name("error").value(false);
+                            printList(out, true);
+                        } else {
+                            out.name("error").value(true);
+                            out.name("reason").value("Already stopped!");
+                        }
+                        break;
+                    case "info":
+                        if (scanning) {
+                            out.name("error").value(false);
+                            printList(out, false);
+                        } else {
+                            out.name("error").value(true);
+                            out.name("reason").value("Scan is not running!");
+                        }
+                        break;
+                    default:
+                        out.name("error").value(true);
+                        out.name("reason").value("Invalid option! Choose one from [start, stop, info]");
+                        break;
+                }
+                out.endObject();
+            }
+        });
+
+
+    }
+
+    private static void printList(JsonWriter out, boolean clear) {
```

Comment:
nit: This method doesn't actually print. `writeDiscoveredDevices`, something like that?

(If you accept my suggested API change, you should remove `boolean clear` and clear unconditionally after calling this in the `dump`/`poll` handler.)